ovn-tester/tests/service_route.py

```diff
@@ -0,0 +1,82 @@
+from collections import namedtuple
+from ovn_context import Context
+from ovn_workload import Namespace
+from ovn_ext_cmd import ExtCmd
+
+import netaddr
+import ovn_load_balancer as lb
+
+ServiceRouteCfg = namedtuple('ServiceRouteCfg',
+                             ['n_lb_cluster',
+                              'n_lb_node'])
+
+
+DEFAULT_VIP_SUBNET = netaddr.IPNetwork('90.0.0.0/16')
+DEFAULT_VIP_SUBNET6 = netaddr.IPNetwork('9::/32')
+DEFAULT_VIP_PORT = 80
+DEFAULT_BACKEND_PORT = 8080
+
+
+class ServiceRoute(ExtCmd):
+    def __init__(self, config, central_node, worker_nodes, global_cfg):
+        super(ServiceRoute, self).__init__(
+                config, central_node, worker_nodes)
+        test_config = config.get('service_route', dict())
+        self.config = ServiceRouteCfg(
+            n_lb_cluster=test_config.get('n_lb_cluster', 16),
+            n_lb_node=test_config.get('n_lb_node', 4)
+        )
+
+    def provide_vip(self, ip, vip_subnet, index):
+        vip_net = vip_subnet.next(index)
+        vip_ip = vip_net.ip.__add__(1)
+
+        prefix = '[' if vip_subnet.version == 6 else ''
+        suffix = ']' if vip_subnet.version == 6 else ''
+
+        vips = {
+            f'{prefix}{vip_ip}{suffix}:{DEFAULT_VIP_PORT}':
+            [f'{prefix}{ip}{suffix}:{DEFAULT_BACKEND_PORT}']
+        }
+
+        return vips
+
+    def provide_node_load_balancer(self, name, cluster, node,
+                                   ip, vip_subnet, index):
+        load_balancer = lb.OvnLoadBalancer(name, cluster.nbctl)
+        load_balancer.add_vips(self.provide_vip(ip, vip_subnet, index))
```

Comment:
A "node-load balancer" should use as VIP the node's IP address.  So no need to pass the `vip_subnet` argument.  We should instead get the IP address from the WorkerNode.ext_rp member.  That's the router port connecting to the external network.  The problem is today `lr_port_add()` doesn't store the IP/subnet in the resulting `LRPort` tuple.  I think we should change it to do that.

Reply:
@dceara ack, I will fix it

Reply:
I added some more comments on the newest version.